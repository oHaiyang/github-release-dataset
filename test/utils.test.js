/* eslint-env jest */
import {
  buildCodeBlock,
  buildDatasetObj,
  readDatasets,
  insertIntoNote,
  COMMENT,
  DATASET_KEY,
} from '../src/utils';

const _jerryDatasetName = '_jerry';
const _mockNote = `
this is a test line.

\`\`\`yaml
this is not a dataset
\`\`\`

\`\`\`yaml
dataset-name: _jerry
dataset:
  _foe: _tom
\`\`\`

yohohoho

\`\`\`yaml
dataset-name: _tom
dataset:
  _food: _jerry
\`\`\`

this is another test line
`;

describe('buildCodeBlock', () => {
  it('should return a yaml block', () => {
    const _text = '---';

    const _block = `\`\`\`yaml
${COMMENT}

---
\`\`\``;

    expect(buildCodeBlock(_text)).toBe(_block);
  });
});

describe('buildDatasetObj', () => {
  it('should consturct a dataset object', () => {
    const _name = '_jerry';
    const _dataset = {
      _foe: '_tom',
    };

    expect(buildDatasetObj(_name, _dataset)).toEqual({
      [DATASET_KEY]: _name,
      dataset: _dataset,
    });
  });
});

describe('readDatasets', () => {
  it('should read all datasets when datasetName not provided', () => {
    expect(readDatasets(_mockNote)).toMatchSnapshot();
  });

  it('should read datasets have specified name', () => {
    expect(readDatasets(_mockNote, _jerryDatasetName)).toMatchSnapshot();
  });
});

describe('insertIntoNote', () => {
  const note = '_note';
  const text = '_text';

  it('should insert and defaultly insert to bottom', () => {
    expect(insertIntoNote(note, text)).toBe(`${note}\n${text}`);
    expect(insertIntoNote(note, text, false)).toBe(`${note}\n${text}`);
  });

  it('should insert to top', () => {
    expect(insertIntoNote(note, text, true)).toBe(`${text}\n${note}`);
  });
});
