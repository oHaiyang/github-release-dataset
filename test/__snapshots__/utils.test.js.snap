// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseDatasets should read all datasets when datasetName not provided 1`] = `
Array [
  Object {
    "body": "dataset-name: _jerry
dataset:
  _foe: _tom",
    "dataset": Object {
      "_foe": "_tom",
    },
    "datasetObj": Object {
      "dataset": Object {
        "_foe": "_tom",
      },
      "dataset-name": "_jerry",
    },
    "length": 54,
    "name": "_jerry",
    "start": 58,
  },
  Object {
    "body": "dataset-name: _tom
dataset:
  _food: _jerry",
    "dataset": Object {
      "_food": "_jerry",
    },
    "datasetObj": Object {
      "dataset": Object {
        "_food": "_jerry",
      },
      "dataset-name": "_tom",
    },
    "length": 55,
    "name": "_tom",
    "start": 124,
  },
]
`;

exports[`parseDatasets should read datasets have specified name 1`] = `
Array [
  Object {
    "body": "dataset-name: _jerry
dataset:
  _foe: _tom",
    "dataset": Object {
      "_foe": "_tom",
    },
    "datasetObj": Object {
      "dataset": Object {
        "_foe": "_tom",
      },
      "dataset-name": "_jerry",
    },
    "length": 54,
    "name": "_jerry",
    "start": 58,
  },
]
`;
